### PR TITLE
add GHA test.yml, codeql, and dependabot.yml

### PR DESCRIPTION
**Pull request by Xiangs18**

### CODE REVIEW

File: .github/codeql.yml
Comment (MrCreosote) [inline .github/codeql.yml:1]: I assume everything except the test script are the standard GHA scripts?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      fail-fast: false
+      matrix:
+        include:
+          - python-version: '3.7'
+            mongo-version: '3.6'
+          - python-version: '3.7'
+            mongo-version: '7.0.4'
+    services:
+      mongo:
+        image: mongo:${{matrix.python-version}}
```
Comment (MrCreosote) [inline .github/workflows/test.yml]:
```suggestion
        image: mongo:${{matrix.mongo-version}}
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        ports:
+            - 27017:27017
+        options: --name mongo${{matrix.python-version}}
```
Comment (MrCreosote) [inline .github/workflows/test.yml]:
```suggestion
        options: --name mongo${{matrix.mongo-version}}
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+          # set HOMEDIR
+          export HOMEDIR=`pwd`
+
+          # move to parent dir to install binaries etc
+          cd ..
+
+          # setup kb-sdk
+          mkdir -p $(pwd)/bin
+          docker run kbase/kb-sdk genscript > $(pwd)/bin/kb-sdk
```
Comment (MrCreosote) [inline .github/workflows/test.yml]: This is the old sdk; I'm surprised it works here

File: lib/biokbase/catalog/registrar.py [hunk 4]
```diff
@@ -252,7 +252,7 @@ def sanity_checks_and_parse(self, basedir, git_commit_hash):
         with codecs.open(os.path.join(basedir, yaml_filename), 'r', "utf-8",
                          errors='ignore') as kb_yaml_file:
             kb_yaml_string = kb_yaml_file.read()
-        self.kb_yaml = yaml.load(kb_yaml_string)
+        self.kb_yaml = yaml.load(kb_yaml_string, Loader=yaml.FullLoader)
```
Comment (MrCreosote) [inline lib/biokbase/catalog/registrar.py]: Does this have any security concerns? IIRC there's a safe loader that is less subject to problems

File: test/local_function_module_test.py
Comment (MrCreosote) [inline test/local_function_module_test.py:1]: Are the changes in this file from the module updates we did? I forget where they came from

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: Install dependencies and set up test config
+        shell: bash
+        env:
+          KBASE_CI_TOKEN: ${{ secrets.KBASE_CI_TOKEN }}
+          ADMIN_USER: kbase_bot
```
Comment (MrCreosote) [inline .github/workflows/test.yml]: This shouldn't be hardcoded - check with one of the sysadmins to see if they can add an env var or secret for the token username

File: lib/biokbase/catalog/registrar.py (hunk 4 above)
Comment (MrCreosote) [reply lib/biokbase/catalog/registrar.py]: I'm not sure why it's needed, the kbase.yml file is very simple

Comment (Xiangs18) [reply lib/biokbase/catalog/registrar.py]: TypeError: load() missing 1 required positional argument: 'Loader'
BUILD_ERROR: load() missing 1 required positional argument: 'Loader'

https://github.com/kbase/catalog/actions/runs/12307585559/job/34351499135?pr=136

File: test/local_function_module_test.py
Comment (Xiangs18) [reply test/local_function_module_test.py:1]: https://kbase.slack.com/archives/D04UCJJTUR1/p1714607433758529

File: .github/codeql.yml
Comment (Xiangs18) [reply .github/codeql.yml:1]: correct

File: .github/workflows/test.yml (hunk 1 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: 👍

File: .github/workflows/test.yml (hunk 2 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: 👍

File: .github/workflows/test.yml (hunk 5 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: 👍

File: .github/workflows/test.yml (hunk 3 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: It is similar to https://kbase.github.io/kb_sdk_docs/tutorial/2_install.html

For setting up, how is the new SDK different from the old one?

Comment (MrCreosote) [reply .github/workflows/test.yml]: Yeah, that tutorial probably won't work in a lot of scenarios

The SDK patch Dakota made upgrades the docker service

**Comment by Xiangs18** [comment]
Mongo7 upgrade PR next

File: .github/codeql.yml
Comment (MrCreosote) [reply .github/codeql.yml:1]: :+1:

File: lib/biokbase/catalog/registrar.py (hunk 4 above)
Comment (MrCreosote) [reply lib/biokbase/catalog/registrar.py]: The question isn't whether there's a loader specified at all, it's whether it needs to be a FullLoader

Comment (MrCreosote) [reply lib/biokbase/catalog/registrar.py]: it looks like there's a yaml.safe_load() function as well

File: .github/workflows/test.yml (hunk 1 above)
Comment (MrCreosote) [reply .github/workflows/test.yml]: :+1:

File: .github/workflows/test.yml (hunk 2 above)
Comment (MrCreosote) [reply .github/workflows/test.yml]: :+1:

File: .github/workflows/test.yml (hunk 5 above)
Comment (MrCreosote) [reply .github/workflows/test.yml]: :+1:

File: test/local_function_module_test.py
Comment (MrCreosote) [reply test/local_function_module_test.py:1]: Ok, it was switching from the prior commit to this one https://github.com/kbaseincubator/catalog_test_module/commit/a8915afe6811de9199897d710348befad8f6f7ab

File: lib/biokbase/catalog/registrar.py (hunk 4 above)
Comment (Xiangs18) [reply lib/biokbase/catalog/registrar.py]: 👍

Comment (MrCreosote) [reply lib/biokbase/catalog/registrar.py]: :+1:

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+  catalog_tests:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - python-version: '3.7'
+            mongo-version: '3.6'
+          # - python-version: '3.7'
+          #   mongo-version: '7.0.4'
```
Comment (MrCreosote) [inline .github/workflows/test.yml:27]: is this supposed to be commented out?

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        with:
+          python-version: ${{matrix.python-version}}
+
+      - name: Install dependencies and set up test config
+        shell: bash
+        env:
+          KBASE_CI_TOKEN: ${{ secrets.KBASE_CI_TOKEN }}
+          ADMIN_USER: ${{ secrets.KBASE_BOT_USER_CI }}
+
+        run: |
```
Comment (MrCreosote) [inline .github/workflows/test.yml:52]: Looking at travis.yml, there's ant and jars install steps which I assume are for NMS, which is required for tests. How do the tests pass now without these steps?

File: .github/workflows/test.yml (hunk 6 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:27]: yes, https://github.com/kbase/catalog/pull/136#issuecomment-2546940588

File: .github/workflows/test.yml (hunk 3 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: Is there anywhere I can find the latest setup/install instructions for kb-sdk?

File: .github/workflows/test.yml (hunk 7 above)
Comment (Xiangs18) [reply .github/workflows/test.yml:52]: submodules (jars, kbapi_common, NMS) got pulled recursively in the repo checkout step. No need to install jars repetitively.

https://github.com/kbase/catalog/actions/runs/12362708941/job/34502579817?pr=136#step:3:99

File: .github/workflows/test.yml (hunk 3 above)
Comment (MrCreosote) [reply .github/workflows/test.yml]: https://github.com/kbase/kb_sdk_patch

If it works it works though

File: .github/workflows/test.yml (hunk 7 above)
Comment (MrCreosote) [reply .github/workflows/test.yml:52]: :+1:

**Review by MrCreosote — APPROVED** [review]
lgtm

File: .github/workflows/test.yml (hunk 3 above)
Comment (Xiangs18) [reply .github/workflows/test.yml]: I switched to the patch, and it seems to be working.

**Pull request merged**